employees/tests/test_unit_report_viewset.py

```diff
@@ -148,3 +151,312 @@ def test_report_detail_view_should_delete_report_on_delete(self):
+        request.user = self.user
+        response = ReportDetail.as_view()(request, pk=999)
+        self.assertEqual(response.status_code, 404)
+
+    def test_custom_report_detail_view_should_update_report_on_post(self):
+        new_description = 'Some other description'
+        request = APIRequestFactory().post(
+            path=reverse('custom-report-detail', args=(self.report.pk,)),
+            data={
+                'date': datetime.datetime.now().date(),
+                'description': new_description,
+                'project': self.project,
+                'work_hours': Decimal('8.00'),
+            },
+        )
+        request.user = self.user
+        response = ReportDetail.as_view()(request, pk=self.report.pk)
+        self.report.refresh_from_db()
+        current_description = self.report.description
+        # SELF NOTE: It appears that save() does not override reference on model object (edit is unavailable)
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(current_description, new_description)
+
+    def test_custom_report_detail_view_should_not_update_report_on_discard(self):
+        new_description = 'Some other description'
+        request = APIRequestFactory().post(
+            path=reverse('custom-report-detail', args=(self.report.pk,)),
+            data={
+                'date': datetime.datetime.now().date(),
+                'description': new_description,
+                'project': self.project,
+                'work_hours': Decimal('8.00'),
+                'discard': "Discard"
+            },
+        )
+        request.user = self.user
+        response = ReportDetail.as_view()(request, pk=self.report.pk)
+        self.report.refresh_from_db()
+        current_description = self.report.description
+        # SELF NOTE: It appears that save() does not override reference on model object (edit is unavailable)
+        self.assertEqual(response.status_code, 302)
+        self.assertNotEqual(current_description, new_description)
+
+    def test_custom_report_detail_view_should_not_update_report_on_post_if_form_is_invalid(self):
+        new_description = 'Some other description'
+        request = APIRequestFactory().post(
+            path=reverse('custom-report-detail', args=(self.report.pk,)),
+            data={
+                'description': new_description,
+                'project': self.project,
+                'work_hours': Decimal('8.00'),
+            },
+        )
+        request.user = self.user
+        response = ReportDetail.as_view()(request, pk=self.report.pk)
+        self.report.refresh_from_db()
+        current_description = self.report.description
+        # SELF NOTE: It appears that save() does not override reference on model object (edit is unavailable)
+        self.assertEqual(response.status_code, 200)
+        self.assertIsNotNone(response.data['errors'])
+        self.assertNotEqual(new_description, current_description)
```

Comment:
Please apply everywhere what I wrote above about comment and `current_description` variable

Reply:
Done